src/codecs/html/html.test.ts

```diff
@@ -117,17 +136,23 @@ describe('Encode & Decode cite nodes', () => {
   })
 })
 
-describe('Encode & Decode code-expression nodes', () => {
-  const codeExpressionSchema = codeExpression({
+describe('Encode & Decode CodeExpression nodes', () => {
+  const codeExpressionNode = codeExpression({
     text: 'x * 2',
     output: '42',
     programmingLanguage: 'python'
   })
-  const codeExpressionHTML =
-    '<stencila-code-expression programming-language="python"><code slot="text">x * 2</code><output slot="output">42</output></stencila-code-expression>'
+  const codeExpressionHTML = `
+    <stencila-code-expression
+      itemtype="${encodeMicrodataItemtype('CodeExpression')}"
```

Comment:
_Posting this ahead of full review:_
I would like us to move away from using string values as much as possible (here and elsewhere), and instead use Enums or simple object key/value pairs to make any updates easier and prevent mistakes.

Reply:
Agreed. In the case of `encodeMicrodataItemtype` the parameter is `keyof Types` so Typescript will indeed catch errors:

```
    src/codecs/html/html.test.ts:147:43 - error TS2345: Argument of type '"CodeExpressio"' is not assignable to parameter of type '"Article" | "Cite" | "CiteGroup" | "Code" | "CodeBlock" | "CodeChunk" | "CodeExpression" | "Collection" | "CreativeWork" | "Datatable" | "Organization" | "Periodical" | "PublicationIssue" | ... 69 more ... | "VariableTypes"'.

    147       itemtype="${encodeMicrodataItemtype('CodeExpressio')}"
```

For `encodeMicrodataProperty` the parameter is a currently a string since we don't yet have an enumeration of all the properties across all types (but presumably we could make that?)